blog/_posts/2019-08-20-multi-container-live-update.md

```diff
@@ -0,0 +1,77 @@
+---
+slug: multi-container-live-update
+date: 2019-08-20
+author: maia
+layout: blog
+title: "Live Update Multiple Containers Per Pod"
+subtitle: "Because your sidecar deserves iteration too!"
+image: sidecarDog.jpg
+image_caption: "Who WOULDN'T want to iterate on this sidecar?!"
+tags:
+  - docker
+  - kubernetes
+  - microservices
+  - tilt
+  - containers
+keywords:
+  - live-update
+  - terminal
+  - tilt
+  - development
+---
+Back in the Dark Ages (i.e. last week), the lion’s share of Tilt’s logic relied on the assumption that there was only one container per pod that was worth caring about. This was the container that we got restart counts from, that we monitored for stale code, and that we updated in place. If you were running your code on _more than one_ container, well, we could only Live Update[^1] the first one, because that was the only one we knew about. Any changes to the other containers would result in a full `docker build && kubectl apply`.
+
+But, as the Bard said: "There are more workflows in heaven and Earth, Horatio, than are dreamt of in your initial product spec." (Or something.) So, we got cracking on some new functionality: as of [Tilt v0.10.0](https://github.com/windmilleng/tilt/releases), you can use Live Update on as many containers per pod as you want! This change didn’t change much on the surface of Tilt, but involved a bunch of restructuring and rewriting under the hood--we had to change a bunch of internal assumptions to account for multiple multiple per pod. This touched everything from logs (`kubectl get logs podabc -c mycontainer`) to resource readiness checks to health monitoring--but we think it was worth it!
```

Comment:
maybe "This change didn't change" -> "This feature didn't change"

chaaaaaaange

Reply:
lol gooood catch 😳